docs: scoping the AWS IAM policy to explicitely defined AWS Route53 zones (#5663)
kubernetes-sigs/external-dns: docs/tutorials/aws.md

our example, we'll call the policy `AllowExternalDNSUpdates` (but you can call
it whatever you prefer).

> If you prefer, you may fine-tune the policy to permit updates only to explicit

Reviewer: not clear the reason why we removing this msg

our example, we'll call the policy `AllowExternalDNSUpdates` (but you can call
it whatever you prefer).

```json
{
  "Version": "2012-10-17",
  "Statement": [
    {
      "Effect": "Allow",
      "Action": [
        "route53:ChangeResourceRecordSets",
        "route53:ListResourceRecordSets",
        "route53:ListTagsForResources"
      ],
      "Resource": [

>         "arn:aws:route53:::hostedzone/<INSERT YOUR ZONE ID HERE>"

Reviewer 2: ```suggestion
        "arn:aws:route53:::hostedzone/*"
```

    {
      "Effect": "Allow",
      "Action": [
        "route53:ChangeResourceRecordSets",
        "route53:ListResourceRecordSets",
        "route53:ListTagsForResources"
      ],
      "Resource": [

>         "arn:aws:route53:::hostedzone/<INSERT YOUR ZONE ID HERE>"

Reviewer 2: ```suggestion
        "arn:aws:route53:::hostedzone/*"
```

    {
      "Effect": "Allow",
      "Action": [
        "route53:ChangeResourceRecordSets",
        "route53:ListResourceRecordSets",

>         "route53:ListTagsForResources"

Reviewer: This will not work with service specific conditions.

Reviewer: ListTagsForResources will not work as it will require healthcheck resource id as well.

Author: should then `ListTagsForResources` be placed under a separate allow statement? 
this way it will possible to fine tune it later passing either zone(s) or health check(s) to it

}
```

### Further improvements

> Both policies can be further enhanced by tightening them down. Following the [principle of least privelege](https://docs.aws.amazon.com/IAM/latest/UserGuide/best-practices.html#grant-least-privilege) and explicitely providing a list of selected zones instead of `*` you can scope the deployment down allowing changes only to zones from the list hence reducing the blast radius and improving auditability.

Reviewer 2: ```suggestion
Both policies can be further enhanced by tightening them down. Following the [principle of least privilege](https://docs.aws.amazon.com/IAM/latest/UserGuide/best-practices.html#grant-least-privilege) and explicitly providing a list of selected zones instead of `*` you can scope the deployment down allowing changes only to zones from the list hence reducing the blast radius and improving auditability.
```
